### PR TITLE
arash/m-tc

### PR DESCRIPTION
**Pull request by arash-fs**
# Description
Make T&C page responsive for mobile view
Fixes # (issue)
T&C page
## Type of change

Please delete options that are not relevant.
-   [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: src/components/elements/side-tab.js [hunk 1]
```diff
@@ -2,26 +2,38 @@ import React, { useEffect, useState } from 'react'
 import PropTypes from 'prop-types'
 import styled from 'styled-components'
 import { navigate } from '@reach/router'
+import MediaQuery from 'react-responsive'
```
Comment (mustofa-binary) [inline src/components/elements/side-tab.js]: you can change this to use `show` component

File: src/components/elements/side-tab.js [hunk 2]
```diff
@@ -36,15 +48,35 @@ const StyledTab = styled.li`
...
+        color: var(--color-red);
+        font-size: 2rem;
+    }
+`
+const ArrowWrapper = styled(Arrow)`
+    transform: ${props =>
+        props.what_is_active_tab === '-' ? 'rotate(0deg)' : 'rotate(180deg)'};
+`
+const Tab = ({ active_tab, label, onClick, text, mobile }) => {
     const className = active_tab === label ? 'tab-active' : ''
```
Comment (mustofa-binary) [inline src/components/elements/side-tab.js]: and yes, i guess, we can make this into class component instead as you suggested

File: src/images/svg/arrow-1.svg [hunk 3]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="16px" height="16px" viewBox="0 0 16 16" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
```
Comment (mustofa-binary) [inline src/images/svg/arrow-1.svg]: aria labelled by and id title

File: src/components/elements/side-tab.js [hunk 4]
```diff
@@ -5,23 +5,35 @@ import { navigate } from '@reach/router'
...
+        width: 100%;
+    }
 `
 
 const TabContent = styled.div`
     flex: 1;
 `
 
 const StyledTab = styled.li`
-    cursor: pointer;
```
Comment (mustofa-binary) [inline src/components/elements/side-tab.js]: why are we removing cursor?

File: src/components/elements/side-tab.js [hunk 5]
```diff
@@ -36,15 +48,35 @@ const StyledTab = styled.li`
...
-const Tab = ({ active_tab, label, onClick, text }) => {
+    p {
+        font-size: var(--text-size-s);
+        color: var(--color-red);
+        font-size: 2rem;
+    }
+`
+const PathWrapper = styled(Path)`
+    transform: ${props =>
+        props.what_is_active_tab === '-' ? 'rotate(0deg)' : 'rotate(180deg)'};
```
Comment (mustofa-binary) [inline src/components/elements/side-tab.js]: `current_active_tab`?

File: src/components/elements/side-tab.js [hunk 6]
```diff
@@ -61,12 +94,22 @@ function useTabs(initial_active_tab = '', has_hash_routing) {
         if (has_hash_routing) navigate(`#${tab}`)
     }
 
-    return [active_tab, setTab]
+    return [active_tab, setTab, last_active_tab, setLastActiveTab]
```
Comment (mustofa-binary) [inline src/components/elements/side-tab.js]: you mean, previous tab?

File: src/components/elements/side-tab.js (hunk 4 above)
Comment (arash-fs) [reply src/components/elements/side-tab.js]: I can't remember why I removed it :D
I'll turn it back.

File: src/components/elements/side-tab.js (hunk 5 above)
Comment (arash-fs) [reply src/components/elements/side-tab.js]: `current_acctive_tab` contains  `active_tab` obejct. so we can access the properties for example `text` and we can use it for conditions or rendering the props.

File: src/components/elements/side-tab.js (hunk 6 above)
Comment (arash-fs) [reply src/components/elements/side-tab.js]: yes, but I'm gonna change the name to be more understandable.

File: src/components/elements/side-tab.js [hunk 7]
```diff
@@ -5,23 +5,36 @@ import { navigate } from '@reach/router'
...
+    }
 `
 
 const TabContent = styled.div`
     flex: 1;
 `
 
 const StyledTab = styled.li`
-    cursor: pointer;
+    cursor: pionter;
```
Comment (oskar-binary) [inline src/components/elements/side-tab.js]: spelling

```suggestion
    cursor: pointer;
```

File: src/components/elements/side-tab.js [hunk 8]
```diff
@@ -61,12 +98,22 @@ function useTabs(initial_active_tab = '', has_hash_routing) {
...
-    return [active_tab, setTab]
+    return [active_tab, setTab, previous_tab, setLastActiveTab]
 }
 
 const SideTab = ({ children, has_hash_routing }) => {
-    const first_tab = children[0].props.label
-    const [active_tab, setTab] = useTabs(first_tab, has_hash_routing)
+    // we should check the window because When building, Gatsby renders these components on the server where window is not defined.
+    const first_tab =
+        typeof window !== 'undefined'
```
Comment (oskar-binary) [inline src/components/elements/side-tab.js]: we have a `hasWindow` function in utils

File: src/images/svg/path.svg [hunk 9]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<svg width="16px" height="16px" viewBox="0 0 16 16" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" aria-labelledby="path-icon-title">
+    <!-- Generator: Sketch 56.3 (81716) - https://sketch.com -->
+    <title id="path-icon-title">Path icon</title>
```
Comment (oskar-binary) [inline src/images/svg/path.svg]: Rename this to chevron? since it's for the  downward facing `>`

File: src/components/elements/side-tab.js [hunk 10]
```diff
@@ -75,22 +122,61 @@ const SideTab = ({ children, has_hash_routing }) => {
...
+                <Mobile>
+                    <StyledDropDown onClick={handleReset}>
+                        {current_active_tab ? (
+                            <StyledActiveTabText>
+                                {current_active_tab.props.text}
+                            </StyledActiveTabText>
+                        ) : (
+                            <StyledActiveTabText>-</StyledActiveTabText>
+                        )}
+                        <PathWrapper what_is_active_tab={active_tab} />
```
Comment (oskar-binary) [inline src/components/elements/side-tab.js]:
```suggestion
                        <PathWrapper active_tab={active_tab} />
```

File: src/components/elements/side-tab.js (hunk 8 above)
Comment (arash-fs) [reply src/components/elements/side-tab.js]: Done

File: src/components/elements/side-tab.js (hunk 7 above)
Comment (arash-fs) [reply src/components/elements/side-tab.js]: Done

File: src/images/svg/path.svg (hunk 9 above)
Comment (arash-fs) [reply src/images/svg/path.svg]: Done

File: src/components/elements/side-tab.js (hunk 10 above)
Comment (arash-fs) [reply src/components/elements/side-tab.js]: Done

**Pull request merged**